src/coreclr/jit/instrsxarch.h

```diff
@@ -630,8 +630,8 @@ INST3(LAST_BMI_INSTRUCTION, "LAST_BMI_INSTRUCTION", IUM_WR, BAD_CODE, BAD_CODE,
 INST3(LAST_AVX_INSTRUCTION, "LAST_AVX_INSTRUCTION", IUM_WR, BAD_CODE, BAD_CODE, BAD_CODE, INS_TT_NONE, INS_FLAGS_None)
 
 INST3(FIRST_AVX512_INSTRUCTION, "FIRST_AVX512_INSTRUCTION", IUM_WR, BAD_CODE, BAD_CODE, BAD_CODE, INS_TT_NONE, INS_FLAGS_None)
-INST3(movdqu8,           "movdqu8",           IUM_WR, SSEFLT(0x7F), BAD_CODE,     SSEFLT(0x6F),                            INS_TT_FULL_MEM,    INS_FLAGS_None)
-INST3(movdqu16,          "movdqu16",          IUM_WR, SSEFLT(0x7F), BAD_CODE,     SSEFLT(0x6F),                            INS_TT_FULL_MEM,    INS_FLAGS_None)
+INST3(movdqu8,           "movdqu8",           IUM_WR, SSEFLT(0x7F), BAD_CODE,     SSEFLT(0x6F),                            INS_TT_FULL_MEM,    INS_FLAGS_None) // TODO-XARCH-AVX512 : Make this available only when AVX512BW is supported.
+INST3(movdqu16,          "movdqu16",          IUM_WR, SSEFLT(0x7F), BAD_CODE,     SSEFLT(0x6F),                            INS_TT_FULL_MEM,    INS_FLAGS_None) // TODO-XARCH-AVX512 : Make this available only when AVX512BW is supported.
```

Comment:
Why can't these be only when AVX512BW is available right now?

The semantics between `movdqu8/16/32/64` are really only important when `kmask` is involved, when there is no mask they all function identically.

Reply:
It should be possible to have this work much like `vinsertf128` vs `vinserti128`

Where we prefer the more accurate `vinserti128` when AVX2 is available, but since it functions identically to `vinsertf128` we can fallback to just the latter that if only AVX is available

Reply:
That's a solution I'm thinking about. The difference is 'type' doesn't matter in that case. If AVX2 is supported we go to  `vinserti128` and else we know AVX will be supported and we can use `vinsertf128`.(https://github.com/dotnet/runtime/blob/1c4cc74daac582b9276a6a848b6a9eb1fc3a01ed/src/coreclr/jit/hwintrinsiclistxarch.h#L609)

But here, we know AVX512F supported but just for byte and word version, we need to **skip the import** and use software fallback if AVX512BW is NOT supported

https://github.com/dotnet/runtime/pull/80960/files#diff-1aee186c93061d785001e1f5fae74c006728820b497584255e1ec516a16953d0R228.

Reply:
What I'm trying to get at, is that the decision making tree is a little more complicated here.
For Vector256_ we do the following steps
1: `HWIntrinsicInfo::lookupId` : https://github.com/dotnet/runtime/blob/1c4cc74daac582b9276a6a848b6a9eb1fc3a01ed/src/coreclr/jit/hwintrinsic.cpp#L380-L394 and we opt out here for avx2 only https://github.com/dotnet/runtime/blob/1c4cc74daac582b9276a6a848b6a9eb1fc3a01ed/src/coreclr/jit/hwintrinsic.cpp#L427-L430C14

2. While lowering the node for special cases, we choose between avx only vs avx2 depending on availability. Eg : https://github.com/dotnet/runtime/blob/07f10e789f06e8ab94dbbe8c870858ab2eb69cf9/src/coreclr/jit/lowerxarch.cpp#L2133-L2177


I do not know how that can be extended for Vector512 sconsidering something like the following where Intrinsisc might or not be available based on type - https://www.felixcloutier.com/x86/movdqu:vmovdqu8:vmovdqu16:vmovdqu32:vmovdqu64

Reply:
Catch he is *if we need to use the software fallback, it needs to be identified in step 1 at `HWIntrinsicInfo::lookupId` So for eg, incase of `Vector256.ToVector512Unsafe` , this would mean possibly checking the arguments type at lookup which we do not use currently. Alternative is to **find an avx512f only instruction for everything to use when only avx512F is available**

Let me know if there is an easy solution here I'm missing